src/Blueprint/Blueprint/Cmdlets/NewAzureRMBlueprintAssignment.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.Azure.Commands.Blueprint.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Blueprint.Models;
+using System;
+using System.Collections;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using Microsoft.Azure.PowerShell.Cmdlets.Blueprint.Properties;
+using ParameterSetNames = Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants.ParameterSetNames;
+using ParameterHelpMessages = Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants.ParameterHelpMessages;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Commands.Blueprint.Common;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Commands.Blueprint.Cmdlets
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "BlueprintAssignment", SupportsShouldProcess = true)]
+    public class NewAzureRmBlueprintAssignment : BlueprintCmdletBase
+    {
+        #region Parameters
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.BlueprintAssignmentName)]
+        [ValidatePattern("^[0-9a-zA-Z_-]*$", Options = RegexOptions.Compiled | RegexOptions.CultureInvariant)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.BlueprintObject)]
+        [ValidateNotNull]
+        public PSBlueprintBase Blueprint { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.SubscriptionIdToAssign)]
+        [ValidateNotNullOrEmpty]
+        public string[] SubscriptionId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.Location)]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.Batch/operations")]
```

Comment:
>Microsoft.Batch/operations [](start = 28, length = 26)

Is that intended to be that resource type? It's not one of ours.

Reply:
This is a recommended argument completer if you have a "location" parameter in the cmdlets. [https://github.com/Azure/azure-powershell/blob/89eea9c118fa45f05f04e5dd09dc285f4a70a22e/documentation/development-docs/azure-powershell-developer-guide.md#location-completer](https://github.com/Azure/azure-powershell/blob/89eea9c118fa45f05f04e5dd09dc285f4a70a22e/documentation/development-docs/azure-powershell-developer-guide.md#location-completer) 

---
In reply to: [257119814](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257119814) [](ancestors = 257119814)

Reply:
The completer probably queries /providers/{namespace} in ARM that looks at the resource type to figure out allowed location. Can you try it with blueprint assignment resource type first. If it doesn't work right (our manifest says "" for location), we should switch to something from Microsoft.Resources namespace instead. That's guaranteed to be updated when new regions show up.

---
In reply to: [257331542](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257331542) [](ancestors = 257331542,257119814)

Reply:
I see. I updated it to use  Microsoft.Resources namespace with resourceGroups resource type since I saw this being used in other cmdlets as well.

---
In reply to: [257382175](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257382175) [](ancestors = 257382175,257331542,257119814)